File: perf/fib_recursive.c
#include <stdio.h>

static int fib(int n) {
    if (n <= 1) return 1;
    return fib(n - 1) + fib(n - 2);
}

void fibonacci_recursive(int n) {
    for (int i = 1; i < n; i++) {
        fib(i);
    }
}

int main(void) {
    printf("Calculate Fibonacci numbers recursively\n");
    fibonacci_recursive((1 << 5) + (1 << 2) + 2 + 1);
    return 0;
}

File: perf/fib_recursive.py
def fibonacci_recursive(n):
    def fibonacci(n):
        if n <= 1:
            return 1
        else:
            return fibonacci(n - 1) + fibonacci(n - 2)
    for i in range(1, n):
        fibonacci(i)


def main():
    print('Calculate Fibonacci numbers recursively')
    fibonacci_recursive(2**5 + 2**2 + 2 + 1)


main()
